frontend/components/ThemeProvider/Theme.ts

```diff
@@ -0,0 +1,160 @@
+interface IGeneralProperty {
+  body: string;
+  text: string;
+}
+
+interface IButton<T> {
+  default: T;
+  hover: T;
+}
+
+interface IStatusButton<T> {
+  success: T;
+  warning: T;
+  danger: T;
+}
+
+export interface ITheme {
+  body: string;
+  text: string;
+  border: string;
+  svg: {
+    fill: string;
+  };
+  section: {
+    body: string;
+    text: string;
+  };
+  button: IStatusButton<IButton<IGeneralProperty>>;
```

Comment:
Instead of X<Y<Z>>>;  Because these are only used here, can't we just have a nested type? Or, better yet, I noticed ITheme utilises all of this. I'd probably make a class and just do new Theme(values); // if no values, have it default to "" in the constructor.  That way we won't have const dark: ITheme & const light: ITheme. We'll just have new Theme(...values);